lib/src/list_mutations.dart

```diff
@@ -172,14 +176,39 @@ SourceEdit _insertInBlockList(
 
   if (index == list.length) return _appendToBlockList(yamlEdit, list, item);
 
-  final formattedValue = _formatNewBlock(yamlEdit, list, item);
+  var (formattedValue, indent) = _formatNewBlock(yamlEdit, list, item);
 
   final currNode = list.nodes[index];
   final currNodeStart = currNode.span.start.offset;
   final yaml = yamlEdit.toString();
-  final start = yaml.lastIndexOf('\n', currNodeStart) + 1;
 
-  return SourceEdit(start, 0, formattedValue);
+  var currSequenceOffset = yaml.lastIndexOf('-', currNodeStart - 1);
+
+  final (isNested, offset) = _isNestedInBlockList(currSequenceOffset - 1, yaml);
+
+  /// We have to get rid of the left indentation applied by default
+  if (isNested && index == 0) {
+    // Give the previous first element its indent
+    formattedValue = formattedValue.trimLeft() + indent;
+  }
+
+  return SourceEdit(offset, 0, formattedValue);
+}
+
+/// Checks if the [YamlNode]'s list is directly nested within another list
+(bool isNested, int offset) _isNestedInBlockList(int start, String yaml) {
```

Comment:
Are we sure it's a good idea to make this a function? are we going to use it elsewhere?

I don't mind, but if we do, we probably should document it clearly.

Like what is `start` what is `yaml` and what is the list being talked about.

The documentation has to explain that `start` is expected to be the offsets of the `-` indicator for an element in a list formatting in block-mode, and `yaml` is the entire YAML document.

And this method aims to find out if:
 * The element that has a start offset at `start` is inside a nested block list
 * offset for...

Reply:
Since this will probably only be used inside `_insertInBlockList` we could consider inlining it. Also given how hard it is to explain what `start` really is.

But add documentation comments that explains what it does is also fine.